File: app/src/main/java/com/epigram/android/ui/adapters/MyAdapterPlaceholder.java
package com.epigram.android.ui.adapters;

import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import androidx.recyclerview.widget.RecyclerView;

import com.epigram.android.R;

public class MyAdapterPlaceholder extends RecyclerView.Adapter<MyAdapterPlaceholder.MyViewHolder> {

    private int placeholderCount = 3;

    public void clear() {
        placeholderCount = 0;
        notifyDataSetChanged();
    }

    public static class MyViewHolder extends RecyclerView.ViewHolder{
        public LinearLayout linearLayout;
        public MyViewHolder(LinearLayout l){
            super(l);
            this.linearLayout = l;
        }
    }

    @Override
    public MyAdapterPlaceholder.MyViewHolder onCreateViewHolder(ViewGroup parent, int viewType){
        LinearLayout l = (LinearLayout) LayoutInflater.from(parent.getContext()).inflate(R.layout.element_place_holder, parent, false);
        MyViewHolder vh = new MyViewHolder(l);
        return vh;
    }

    @Override
    public void onBindViewHolder(MyViewHolder holder, int position){
    }

    @Override
    public int getItemCount(){
        return placeholderCount;
    }
}

File: app/src/main/java/com/example/epigram/ui/main/MyAdapterPlaceholder.kt
package com.example.epigram.ui.main

import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.LinearLayout
import androidx.recyclerview.widget.RecyclerView
import com.example.epigram.R
import com.example.epigram.ui.main.MyAdapterPlaceholder.MyViewHolder

class MyAdapterPlaceholder : RecyclerView.Adapter<MyAdapterPlaceholder.MyViewHolder>() {

    private var placeholderCount = 3

    fun clear() {
        placeholderCount = 0
        notifyDataSetChanged()
    }

    class MyViewHolder(var linearLayout: LinearLayout) : RecyclerView.ViewHolder(linearLayout)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyAdapterPlaceholder.MyViewHolder {
        val l =
            LayoutInflater.from(parent.context).inflate(R.layout.element_place_holder, parent, false) as LinearLayout
        return MyViewHolder(l)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {}

    override fun getItemCount(): Int {
        return placeholderCount
    }
}
